### PR TITLE
feat: change the menu to add a note

### PR DESCRIPTION
**Pull request by MatthieuSLR9**
# Description

I changed the way you would add a new note by removing the addNote Screen and simply adding a popup that asks for a name and the visibility is set to private by default.

# How Has This Been Tested?
  
### UI Tests

Updated the UI test for Overview and Foldercontent

### End to end Tests

Changed the end to end to reflect the new way to add notes

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged in downstream modules, no conflicts with main

### CODE REVIEW

File: app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+ * @param onDismiss callback to be invoked when the dialog is dismissed
+ * @param onConfirm callback to be invoked when the user confirms the new name and visibility
+ * @param action the action to be performed (e.g., "Create" or "Rename")
+ * @param oldVisibility the previous visibility of the item (if renaming), defaults to
+ *   [Visibility.PRIVATE]
+ * @param oldName the previous name of the item (if renaming), defaults to an empty string
+ * @param type the type of item (e.g., "Folder" or "Note") displayed in the dialog
+ */
+@Composable
+fun CreationDialog(
```
Comment (LawrenceLeitgib) [inline app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt:91]: Good idea to modularize all of that (evoiding duplicate code)

File: app/src/main/java/com/github/onlynotesswent/ui/overview/FolderContent.kt [hunk 2]
```diff
@@ -152,12 +158,32 @@ fun FolderContentScreen(
                 },
                 action = "Rename",
                 oldName = updatedName,
-                oldVis = folder.value!!.visibility)
+                oldVisibility = folder.value!!.visibility)
+          }
+          if (showCreateNoteDialog) {
+            NoteDialog(
```
Comment (LawrenceLeitgib) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/FolderContent.kt:164]: Good code, this is very is to maintain

File: app/src/main/java/com/github/onlynotesswent/ui/overview/Overview.kt [hunk 3]
```diff
@@ -108,10 +113,31 @@ fun OverviewScreen(
             userViewModel = userViewModel,
             context = context,
             navigationActions = navigationActions)
+        if (showCreateNoteDialog) {
+          NoteDialog(
```
Comment (LawrenceLeitgib) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/Overview.kt:117]: This code is the same as the one in Overview , but I think it's fine to leave it as is. Extracting the common code would make too much modularization, which might only make the code messier.

**Review by LawrenceLeitgib — CHANGES_REQUESTED** [review]
Overall good code but I had some trouble with the UI, the button's text aren't displayed correctly, I think we should fix that before merging
Screenshot_20241125_155826

**Comment by LawrenceLeitgib** [comment]
I also found another bug, that the pop-up "discard unsaved change" still apear even after saving changes it also doesn't appear if I modify something. This bug only appear after creating a new note.
[Screen_recording_20241125_161707.webm](https://github.com/user-attachments/assets/cede902a-67e3-4770-aae3-05590d1ff671)

**Comment by MatthieuSLR9** [comment]
> I found another bug, that the pop-up "discard unsaved change" still apear even after saving changes it also doesn't appear if I modify something. This bug only appear after creating a new note. Also when I creat a new note and save it, the course is still Swent while it should be empty. Also when I click on save after editing the note it should redirect me on overview and not keep me on the page [Screen_recording_20241125_161707.webm](https://github.com/user-attachments/assets/cede902a-67e3-4770-aae3-05590d1ff671)

I did not modify the new UI edit note screen as I did not work on it but I will work on it now as part of my PR to address these issues.


> And also the addNote page is not used anymore so I think we can safely delet it

I kept it just in case we wanted to revert back to it in the future. I also left the tests for it. But I will delete it if you think it is better.

**Review by robertaleG — DISMISSED** [review]

File: app/src/androidTest/java/com/github/onlynotesswent/ui/overview/OverviewTest.kt [hunk 4]
```diff
@@ -223,7 +223,21 @@ class OverviewTest {
...
+    composeTestRule.onNodeWithTag("createNoteOrFolder").performClick()
+    composeTestRule.onNodeWithTag("createNote").performClick()
+
+    composeTestRule.onNodeWithTag("inputNoteName").performTextInput("Note Name")
+    composeTestRule.onNodeWithTag("visibilityButton").assertIsDisplayed().performClick()
+
+    composeTestRule.onNodeWithTag("item--Public").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("item--Friends Only").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("item--Private").assertIsDisplayed().performClick()
+    composeTestRule.onNodeWithTag("confirmNoteAction").assertIsDisplayed()
```
[the remark this answers is not in the document]
Comment (robertaleG) [reply app/src/androidTest/java/com/github/onlynotesswent/ui/overview/OverviewTest.kt:240]: Also don't forget to verify correct calls to navigation actions / view models or repositories are made, 
... to actually test the callbacks

**Comment by Roshan-y** [comment]
> Overall good code but I had some trouble with the UI, the button's text aren't displayed correctly, I think we should fix that before merging Screenshot_20241125_155826
> 
> I found another bug, that the pop-up "discard unsaved change" still apear even after saving changes it also doesn't appear if I modify something. This bug only appear after creating a new note. Also when I creat a new note and save it, the course is still Swent while it should be empty. Also when I click on save after editing the note it should redirect me on overview and not keep me on the page [Screen_recording_20241125_161707.webm](https://github.com/user-attachments/assets/cede902a-67e3-4770-aae3-05590d1ff671)
> 
> And also the addNote page is not used anymore so I think we can safely delet it

Hey I fixed the first bug, it was due to trying to save a default course in documentSnapshotToNote in NoteRepositoryFirestore.kt file. The second one, was a decisions I made. I thought maybe the user would want to add a comment or modify the pdf or markdown after saving the metadata so I made it so that they stay in the page

**Comment by Roshan-y** [comment]
> > I kept it just in case we wanted to revert back to it in the future. I also left the tests for it. But I will delete it if you think it is better.
> 
> it's as you want but I think we can alway goes back to old commit if we want to revert some changes

I removed add note as I think it's unnecessary to keep it

File: app/src/androidTest/java/com/github/onlynotesswent/ui/overview/OverviewTest.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Roshan-y) [reply app/src/androidTest/java/com/github/onlynotesswent/ui/overview/OverviewTest.kt:240]: Yes good catch

**Review by LawrenceLeitgib — APPROVED** [review]
Nice changes and fix to the the discard changes pop up, for the UI bug this will indeed be fixed in another PR as this PR was about add note and not edit note

**Pull request merged**